src/main/java/com/todayhouse/domain/story/application/StoryReplyServiceImpl.java

```diff
@@ -0,0 +1,77 @@
+package com.todayhouse.domain.story.application;
+
+import com.todayhouse.domain.story.dao.StoryReplyRepository;
+import com.todayhouse.domain.story.dao.StoryRepository;
+import com.todayhouse.domain.story.domain.Story;
+import com.todayhouse.domain.story.domain.StoryReply;
+import com.todayhouse.domain.story.dto.reqeust.DeleteReplyRequest;
+import com.todayhouse.domain.story.dto.reqeust.CreateReplyRequest;
+import com.todayhouse.domain.story.dto.response.CreateReplyResponse;
+import com.todayhouse.domain.story.dto.response.ReplyGetResponse;
+import com.todayhouse.domain.story.exception.ReplyNotFoundException;
+import com.todayhouse.domain.user.dao.UserRepository;
+import com.todayhouse.domain.user.domain.User;
+import com.todayhouse.domain.user.exception.UserNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PageableDefault;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Optional;
+
+import static com.todayhouse.global.error.BaseResponseStatus.REPLY_NOT_FOUND;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+public class StoryReplyServiceImpl implements StoryReplyService {
+    private final StoryRepository storyRepository;
+    private final UserRepository userRepository;
+    private final StoryReplyRepository replyRepository;
+
+
+    @Override
+    public void deleteReply(User user, DeleteReplyRequest request) {
+        Optional<StoryReply> byId = replyRepository.findById(request.getStoryId());
+        StoryReply storyReply = byId.orElseThrow(() -> new ReplyNotFoundException(REPLY_NOT_FOUND));
+        user = userRepository.findByEmail(user.getEmail()).orElseThrow(UserNotFoundException::new);
+        if (!user.getId().equals(storyReply.getUser().getId())) {
+            throw new RuntimeException();
+        }
+        replyRepository.delete(storyReply);
+    }
+
+
+    @Override
+    public Page<ReplyGetResponse> findReplies(Long storyId, @PageableDefault Pageable pageable) {
+        PageRequest of = PageRequest.of(0, 10);
+        Page<StoryReply> storyReplies = replyRepository.findByStoryId(storyId, pageable);
+        Page<ReplyGetResponse> map = storyReplies.map(r -> new ReplyGetResponse(r.getId(), r.getContent(), r.getCreatedDate(), r.getUser()));
+        return map;
+    }
+
+    @Override
+    public CreateReplyResponse replyStory(User user, CreateReplyRequest request) {
```

Comment:
```suggestion
    @Override
    public CreateReplyResponse replyStory(User user, CreateReplyRequest request) {
```
https://resilient-923.tistory.com/391
엔티티를 변경하지 않는 service 메소드들에는 @Transactional(readOnly = true) 붙여주는 것이 좋을 것 같아요.